### PR TITLE
Change getindex to getdataframe for sim_inst

### PR DESCRIPTION
**Pull request by lrennels**
This PR handles Issue #572.

Remove the `getindex` method for `SimulationInstance`and replace it with `getdataframe`.  Users will be able to use `getdataframe` to access results of a simulation while we do some thinking on what type we'd like `getindex` to return and how to handle the multidimensionality of scenarios etc.

### CODE REVIEW

File: docs/src/internals/montecarlo.md [hunk 1]
```diff
@@ -163,10 +163,10 @@ If provided, the generated trials and results will be saved in the indicated
 `trials_output_filename` and `results_output_dir` respectively. If `results_in_memory` is set
 to false, then results will be cleared from memory and only stored in the
 `results_output_dir`. After `run`, the results of a `SimulationInstance` can be accessed using
-the familiar `getindex` function with the following signature, which returns a `DataFrame`.
```
Comment (rjplevin) [inline docs/src/internals/montecarlo.md]: Not sure the word "familiar" is still appropriate.

Comment (lrennels) [reply docs/src/internals/montecarlo.md]: Good point I'll change that

**Comment by lrennels** [comment]
@davidanthoff I made these (minimal) changes and now get a strange `getindex` error for a function I did not touch...

When Travis runs `test_main.jl` and uses `set_param!` we get the error:

```
julia> set_param!(x1, :foo1, :par2, [true true false; true false false; true true true])
ERROR: MethodError: no method matching getindex(::OrderedCollections.OrderedDict{Int64,Int64}, ::Int64)
You may have intended to import Base.getindex
Stacktrace:
 [1] getindex(::Mimi.Dimension{Int64}, ::Int64) at /Users/lisarennels/.julia/dev/Mimi/src/core/dimensions.jl:34
 [2] _check_labels(::Mimi.ModelDef, ::Mimi.ComponentDef, ::Symbol, ::Mimi.ArrayModelParameter{Mimi.TimestepArray{Mimi.FixedTimestep{2010,10,LAST} where LAST,Union{Missing, Bool},2,1}}) at /Users/lisarennels/.julia/dev/Mimi/src/core/connections.jl:86
 [3] #connect_param!#21(::Bool, ::typeof(connect_param!), ::Mimi.ModelDef, ::Symbol, ::Symbol, ::Symbol) at /Users/lisarennels/.julia/dev/Mimi/src/core/connections.jl:110
 [4] connect_param! at /Users/lisarennels/.julia/dev/Mimi/src/core/connections.jl:106 [inlined]
 [5] set_param!(::Mimi.ModelDef, ::Symbol, ::Symbol, ::Array{Bool,2}, ::Nothing) at /Users/lisarennels/.julia/dev/Mimi/src/core/defs.jl:512
 [6] set_param! at /Users/lisarennels/.julia/dev/Mimi/src/core/delegate.jl:55 [inlined] (repeats 2 times)
 [7] top-level scope at REPL[47]:1
```

Do you see how my changes could have caused this?

File: src/mcs/mcs_types.jl [hunk 2]
```diff
@@ -225,6 +225,10 @@ struct SimIterator{NT, T}
     end
 end
 
-function Base.getindex(sim_inst::SimulationInstance, comp_name::Symbol, datum_name::Symbol; model::Int = 1)
+function getdataframe(sim_inst::SimulationInstance, comp_name::Symbol, datum_name::Symbol; model::Int = 1)
     return sim_inst.results[model][(comp_name, datum_name)]
 end
+
+function getindex(sim_inst::SimulationInstance, comp_name::Symbol, datum_name::Symbol; model::Int = 1)
+    error("getindex method for `SimulationInstance` has been replaced with getdataframe function for consistency of return type, please use getdataframe instead")
```
Comment (ckingdon95) [inline src/mcs/mcs_types.jl]: @lrennels Shoulr this be `Base.getindex`? maybe this could be related to the error you were getting?

Comment (lrennels) [reply src/mcs/mcs_types.jl]: Oh good call I’ll try that

**Comment by lrennels** [comment]
@davidanthoff I'm getting PyCall errors in this Travis build, any advice?  I tried closing and opening the PR but got the same error.

**Pull request merged**